start.sh

```diff
@@ -116,40 +118,41 @@ else
 fi
 
 cd $gaianet_base_dir
-llamaedge_wasm="$gaianet_base_dir/llama-api-server.wasm"
+llamaedge_wasm="$gaianet_base_dir/rag-api-server.wasm"
 if [ ! -f "$llamaedge_wasm" ]; then
     printf "LlamaEdge wasm not found at $llamaedge_wasm\n"
     exit 1
 fi
 
 # command to start LlamaEdge API Server
 cd $gaianet_base_dir
-cmd="wasmedge --dir .:./dashboard \
+cmd=(wasmedge --dir .:./dashboard \
   --nn-preload default:GGML:AUTO:$chat_model_name \
   --nn-preload embedding:GGML:AUTO:$embedding_model_name \
-  llama-api-server.wasm -p $prompt_type \
+  rag-api-server.wasm \
   --model-name $chat_model_stem,$embedding_model_stem \
   --ctx-size $chat_ctx_size,$embedding_ctx_size \
-  --qdrant-url http://127.0.0.1:6333 \
-  --qdrant-collection-name "default" \
-  --qdrant-limit 3 \
-  --qdrant-score-threshold 0.4 \
+  --prompt-template $prompt_type \
   --web-ui ./ \
   --socket-addr 0.0.0.0:$llamaedge_port \
   --log-prompts \
-  --log-stat"
+  --log-stat)
+
+# Add system prompt if it exists
+if [ -n "$system_prompt" ]; then
+    cmd+=("--system-prompt" "$system_prompt")
```

Comment:
I believe that you should escape this. The current command line looks like this when printed to the console:

```
wasmedge --dir .:./dashboard --nn-preload default:GGML:AUTO:Llama-2-7b-chat-hf-Q5_K_M.gguf --nn-preload embedding:GGML:AUTO:all-MiniLM-L6-v2-ggml-model-f16.gguf rag-api-server.wasm --model-name Llama-2-7b-chat-hf-Q5_K_M,all-MiniLM-L6-v2-ggml-model-f16 --ctx-size 4096,384 --prompt-template llama-2-chat --web-ui ./ --socket-addr 0.0.0.0:8080 --log-prompts --log-stat --system-prompt Use the following pieces of context to answer the user's question.\nIf you don't know the answer, just say that you don't know, don't try to make up an answer.\n----------------\n
```

Reply:
The command is in the form of `array` instead of a string, so we cannot edit it to insert `escape` characters.

Reply:
How does ppl pass several strings to a CLI command as arguments?

Reply:
I mean can we make the argument a quoted string like 

```
--rag-prompt "Please answer based on context below\n"
```

Reply:
Fixed. Thanks!